activerecord/lib/active_record/nested_attributes.rb

```diff
@@ -355,12 +355,18 @@ def accepts_nested_attributes_for(*attr_names)
         options.update(attr_names.extract_options!)
         options.assert_valid_keys(:allow_destroy, :reject_if, :limit, :update_only)
         options[:reject_if] = REJECT_ALL_BLANK_PROC if options[:reject_if] == :all_blank
+        options[:klass] = self
 
         attr_names.each do |association_name|
           if reflection = _reflect_on_association(association_name)
             reflection.autosave = true
             define_autosave_validation_callbacks(reflection)
 
+            association_options = self.nested_attributes_options[association_name.to_sym]
+            if association_options && association_options[:klass] == self
```

Comment:
Since we're never destructuring the `options[:klass]` into its own local variable (like `klass`), would it be appropriate to use `:class` as the Hash key, since there isn't a risk of colliding with the `class` reserved word?

Reply:
Ya true, I did it out of habit, but I think you're right, we can just use `class` here.

Reply:
Actioned here: https://github.com/rails/rails/compare/4cdb1c9ad91ece2ed33c51fd404806b88d4fba61..56f53113a69f5a76fb068abe6e12710bdfed5b3c